src/test/ui/error-codes/E0746.rs

```diff
@@ -10,9 +10,9 @@ fn foo() -> dyn Trait { Struct }
 
 fn bar() -> dyn Trait { //~ ERROR E0746
     if true {
-        return 0;
+        return 0u32;
```

Comment:
why are these changes necessary? What happens if you don't do it?

Reply:
This was needed before I did `resolve_vars_if_possible` because `{integer}` wasnt being detected as implementing `Trait`. Now it does.

Reply:
So these changes can be reverted?

Reply:
yes